### DISCUSSION TITLE
Find data with same vectors

### DISCUSSION THREAD

**Discussion by hueidou** [milvus-io/milvus#35423] [Q&A and General discussion]
There are many data with same vectors in the Milvus library. How can we find these data with completely same vectors?

Just like sql `select eb,count(*) from table group by eb where count(*) > 1`.

**Comment by yhmo** [comment]
Milvus cannot do grouping for vectors.
My solution is:
- Use query iterator to iterate all the vectors from db: https://milvus.io/docs/with-iterators.md#Query-with-an-iterator
- Construct a vector pool to store the distinct vectors
- The iterator returns vectors one by one. For each returned vector, compare it with the vectors in the vector pool. If the returned vector equals any one vector in the pool, skip it. Else, put the returned vector into the vector pool.

You need to define a method to determine whether two vectors are equal, the method should accept a tolerance. If the Euclidean distance between two vectors is smaller than the tolerance, then we say the two vectors are equal.
This solution will take a long time if the number of vectors is large.
